File: Java/remove-vowels.java
/*
* https://leetcode.com/problems/remove-vowels-from-a-string/
* Runtime: 2 ms, faster than 28.65% of Java online submissions for Remove Vowels from a String.
* Memory Usage: 37.6 MB, less than 100.00% of Java online submissions for Remove Vowels from a String.
*/

class Solution {
    public String removeVowels(String S) {
        return S.replaceAll("a|e|i|o|u", "");
    }
}

File: Python3/remove-vowels.py
# https://leetcode.com/problems/remove-vowels-from-a-string/
# Runtime: 24 ms, faster than 85.78% of Python3 online submissions for Remove Vowels from a String.
# Memory Usage: 12.7 MB, less than 100.00% of Python3 online submissions for Remove Vowels from a String.

class Solution:
    def removeVowels(self, S: str) -> str:
        i = 0
        while i < len(S):
            if S[i] == 'a' or S[i] == 'e' or S[i] == 'i' or S[i] == 'o' or S[i] == 'u':
                S = S[:i] + S[i + 1:]
                i = i - 1
            i = i + 1
        return S
